aptible/resource_endpoint.go

```diff
@@ -149,146 +153,274 @@ func resourceEndpointValidate(_ context.Context, diff *schema.ResourceDiff, _ in
 	return err
 }
 
-func resourceEndpointCreate(d *schema.ResourceData, meta interface{}) error {
-	client := meta.(*providerMetadata).LegacyClient
+func resourceEndpointCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	m := meta.(*providerMetadata)
+	legacy := m.LegacyClient
+	client := m.Client
+	ctx = m.APIContext(ctx)
 	service := aptible.Service{}
-	var err error
+	diags := diag.Diagnostics{}
 
 	processType := d.Get("process_type").(string)
 	resourceID := int64(d.Get("resource_id").(int))
 	resourceType := d.Get("resource_type").(string)
-	interfaceSlice := d.Get("ip_filtering").([]interface{})
-	ipWhitelist, _ := aptible.MakeStringSlice(interfaceSlice)
-	interfaceContainerPortsSlice := d.Get("container_ports").([]interface{})
-	containerPorts, _ := aptible.MakeInt64Slice(interfaceContainerPortsSlice)
+	ipWhitelist, err := makeStringSlice(d.Get("ip_filtering").([]interface{}))
+	if err != nil {
+		return append(diags, diag.Diagnostic{
```

Comment:
Isn't diags a struct (diag.Diagnostics{})? How are you appending to it here and below, or did you mean to declare it and the return type as `[]diag.Diagnostics{}` instead?

Reply:
There's both `diag.Diagnostic{}` singular and `diag.Diagnostics{}` plural. The latter acts like a slice of `diag.Diagnostic{}` (I think it's a type alias), hence why `append` works.